### PR TITLE
feat(crons): Allow timeline view to be used with arbitrary start/end

### PR DESCRIPTION
**Pull request by davidenwang**
Allows the timeline view to operate off an arbitrary start/end or period from the page date filter. The main change now is that many of the timeline components previously worked off a `timeWindow`: `1h`, `24h`, etc... which they would use as a key to retrieve a `timeWindowConfig` which held information about how far apart time labels should be, what date info to show in tooltips, etc...

Now we use start/end to generate a config which is also dynamic based on the width of the timeline view. We have a number of preset time intervals to space labels with such as individual minutes, 10 minutes, 30 minutes, etc... going up to `N` number of days between each label. 

This will eventually pave the way for the timeline view to be added onto the monitor details page.

### CODE REVIEW

File: static/app/views/monitors/components/overviewTimeline/gridLines.tsx [hunk 1]
```diff
@@ -4,31 +4,31 @@ import moment from 'moment';
...
+ */
+function alignTimeMarkersToStartOf(date: moment.Moment, timeMarkerInterval: number) {
+  if (timeMarkerInterval < 60) {
+    date.minute(date.minutes() - (date.minutes() % timeMarkerInterval));
+  } else if (timeMarkerInterval < 60 * 24) {
     date.startOf('hour');
+  } else {
+    date.startOf('day');
   }
 }
```
Comment (evanpurkhiser) [inline static/app/views/monitors/components/overviewTimeline/gridLines.tsx:33]: Does this actually work? I thought moment was immutable so you would need to return these?

Comment (davidenwang) [reply static/app/views/monitors/components/overviewTimeline/gridLines.tsx:33]: nah this works out and seems like moments are mutable https://momentjs.com/guides/#/lib-concepts/mutability/

File: static/app/views/monitors/components/overviewTimeline/utils.tsx [hunk 2]
```diff
@@ -2,39 +2,54 @@ import moment from 'moment';
...
+  const timeLabelMinutes = startEndMinutes * (TIMELABEL_WIDTH / timelineWidth);
+
+  for (const minutes of minuteRanges) {
+    if (minutes > timeLabelMinutes) {
+      return {
+        cursorLabelFormat: getFormat({timeOnly: true}),
+        elapsedMinutes: startEndMinutes,
+        timeMarkerInterval: minutes,
+        dateTimeProps: {timeOnly: true},
+        tooltipDateTimeProps: {timeOnly: true},
```
Comment (evanpurkhiser) [inline static/app/views/monitors/components/overviewTimeline/utils.tsx:41]: Let's just have one prop called `dateLabelFormat` and we can use that for the cursor and the tooltip

File: static/app/views/monitors/components/overviewTimeline/utils.tsx [hunk 3]
```diff
@@ -2,39 +2,54 @@ import moment from 'moment';
...
+): TimeWindowOptions {
+  // Acceptable intervals between time labels, in minutes
+  const minuteRanges = [1, 10, 30, 60, 4 * 60, 8 * 60, 12 * 60];
+  const startEndMinutes = (end.getTime() - start.getTime()) / (1000 * 60);
+  const timeLabelMinutes = startEndMinutes * (TIMELABEL_WIDTH / timelineWidth);
+
+  for (const minutes of minuteRanges) {
+    if (minutes > timeLabelMinutes) {
+      return {
+        cursorLabelFormat: getFormat({timeOnly: true}),
```
Comment (evanpurkhiser) [inline static/app/views/monitors/components/overviewTimeline/utils.tsx:37]: Let's bring back the `dateOnly` `timeOnly` and `seconds` logic.

**Review by evanpurkhiser — APPROVED** [review]

**Pull request merged**